codegen/type_resolution.py

```diff
@@ -1091,6 +1115,19 @@ def lookup_named_type(
                 f"typedef {prefix}{format_specialization(specialization)} : ...",
             )
 
+        example_candidate = self._unresolved_typedefs[prefix][0]
```

Comment:
Is this guaranteed to exist?

Reply:
Yup. The previous lines confirm that ```prefix in self._unresolved_typedefs```. Ofc the python type system implies `self._unresolved_typedefs` can be an empty list, but this never happens since we build `self._unresolved_typedefs` by appending typedefs.